great_expectations/data_context/store/expectations_store.py

```diff
@@ -190,20 +112,79 @@ def gx_cloud_response_json_to_object_dict(response_json: Dict) -> Dict:
         return suite_dict
 
     def _add(self, key, value, **kwargs):
+        value = self._add_ids_to_new_objects(value)
         try:
-            return super()._add(key=key, value=value, **kwargs)
+            result = super()._add(key=key, value=value, **kwargs)
+            if self.cloud_mode:
+                result = cast(GXCloudResourceRef, result)
+                value = self._add_cloud_ids_to_local_suite_and_expectations(
+                    local_suite=value,
+                    cloud_suite=result.response["data"]["attributes"]["suite"],
+                )
+            return result
         except gx_exceptions.StoreBackendError:
             raise gx_exceptions.ExpectationSuiteError(
                 f"An ExpectationSuite named {value.expectation_suite_name} already exists."
             )
 
     def _update(self, key, value, **kwargs):
+        value = self._add_ids_to_new_objects(value)
         try:
-            return super()._update(key=key, value=value, **kwargs)
-        except gx_exceptions.StoreBackendError:
+            result = super()._update(key=key, value=value, **kwargs)
+            if self.cloud_mode:
+                result = cast(GXCloudResourceRef, result)
+                value = self._add_cloud_ids_to_local_suite_and_expectations(
+                    local_suite=value,
+                    cloud_suite=result.response["data"]["attributes"]["suite"],
+                )
+        except gx_exceptions.StoreBackendError as exc:
+            # todo: this generic error clobbers more informative errors coming from the store
             raise gx_exceptions.ExpectationSuiteError(
                 f"Could not find an existing ExpectationSuite named {value.expectation_suite_name}."
+            ) from exc
+
+    def _add_ids_to_new_objects(self, suite: ExpectationSuite) -> ExpectationSuite:
+        if self.cloud_mode:
+            # cloud backend creates IDs, and we add them after persistence
+            return suite
+        # ensure suite has an ID
+        if not suite.get("ge_cloud_id"):
+            suite["ge_cloud_id"] = str(uuid.uuid4())
```

Comment:
Still looking at the PR, but I think it's a problem to assign `ge_cloud_id` when we are not in cloud mode.

Reply:
do you know where this is a problem? i was wondering if someone has pinned logic to `ge_cloud_id`, but haven't run across an example of it. it'll be renamed to just `id` in a followup PR.

Reply:
I think it's just a really misleading name at this point. Let's follow up and rename.